### PR TITLE
VS-140: .NET Analyzer stops working after failed analysis

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
Fixed the issue pertaining to VS-140: .NET Analyzer stops working after failed analysis

### CODE REVIEW

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Jira/VS140.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            var filterDefinition = Builders<User>.Filter;
+
+            _ = filterDefinition.Eq(u => u.Name, "User Name1");
+
+            _ = filterDefinition.Lt(u => u.Age, 10) |
+                filterDefinition.Gt(u => u.Age, 20) |
+                (filterDefinition.Ne(u => u.Name, "Bob") &
+                 filterDefinition.Exists(u => u.LastName));
+
+            _ = filterDefinition.Eq(u => u.Scores[1], 100) &
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Jira/VS140.cs:49]: Let's also add a test with full path, like `MongoDB.Driver.Builders<MongoDB.Analyzer.Tests.Common.DataModel.User>.Filter.Eq(...)`

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs [hunk 2]
```diff
@@ -26,6 +26,33 @@ private enum NodeType
         Fluent
     }
 
+    private static SyntaxNode[] GetBuildersDefinitionNodes(SemanticModel semanticModel, SyntaxNode expressionNode)
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs:29]: Done.

File: tests/MongoDB.Analyzer.Tests/Infrastructure/DriverVersionHelper.cs [hunk 3]
```diff
@@ -25,17 +25,17 @@ internal static class DriverVersionHelper
     static DriverVersionHelper()
     {
         var driverVersion = Environment.GetEnvironmentVariable("DRIVER_VERSION");
-        DriverVersions = new[] { NuGetVersion.Parse(driverVersion) };
+        //DriverVersions = new[] { NuGetVersion.Parse(driverVersion) };
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests/Infrastructure/DriverVersionHelper.cs:28]: Done

File: src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs [hunk 4]
```diff
@@ -41,20 +68,22 @@ public static ExpressionsAnalysis ProcessSemanticModel(MongoAnalysisContext cont
...
         {
-            SyntaxNode collectionNode = null;
+            SyntaxNode[] builderDefinitionOrCollectionNodes = null;
             var (nodeType, namedType, expressionNode) = GetNodeType(semanticModel, node);
 
             switch (nodeType)
             {
                 case NodeType.Builders:
                     {
+                        //Get Nodes that represent Builder Definitions
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs:78]: Done

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Jira/VS140.cs [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+                ((Builders<User>.Filter.Lt(u => u.Age, 10)) |
+                 Builders<User>.Filter.Gt(u => u.Age, 20) |
+                (Builders<User>.Filter.Ne(u => u.Name, "Bob") &
+                 Builders<User>.Filter.Exists(u => u.LastName)));
+
+            _ = Builders<User>.Filter.Lt(u => u.Age, 10) |
+                Builders<User>.Filter.Gt(u => u.Age, 20) |
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Jira/VS140.cs:56]: Done

File: src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs [hunk 6]
```diff
@@ -26,6 +26,33 @@ private enum NodeType
...
+            {
+                if (!semanticModel.GetSymbolInfo(node).Symbol.IsDefinedInMongoDriver())
+                {
+                    nodesProcessed.Add(node);
+                }
+                else if (node is MemberAccessExpressionSyntax memberAccessExpressionSyntax)
+                {
+                    var objectExpression = memberAccessExpressionSyntax.Expression;
+
+                    if (semanticModel.GetAliasInfo(objectExpression) != null)
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs:45]: Done

**Pull request closed without merging**